File: app/lib/models/Canon.dart
class Canon {
  final int malId;
  final String title;
  final String url;
  final String imageUrl;
  final String malType;

  Canon({this.malType, this.malId, this.title, this.imageUrl, this.url});

  factory Canon.fromMAL(String malType, Map<String, dynamic> json) {
    if (json == null) return null;
    return Canon(
      malType: malType,
      malId: json['mal_id'],
      title: json['title'],
      imageUrl: json['image_url'],
      url: json['url'],
    );
  }
  factory Canon.fromJson(Map<String, dynamic> json) {
    if (json == null) return null;
    return Canon(
      malType: json["MALType"],
      malId: json['MALID'],
      title: json['Title'],
      url: json['URL'],
      imageUrl: json['ImageURL'],
    );
  }
}

File: srv/models/canons/canon.go
package canons

type Canon struct {
	MALType string
	MALID int
	Title string
	URL string
	ImageURL string
}

func MAL2Canon(malType string, mal map[string]interface{}) Canon {
	return Canon{ 
		MALType: malType,
		MALID: int(mal["mal_id"].(float64)), 
		Title: mal["title"].(string),  
		URL: mal["url"].(string),  
		ImageURL: mal["image_url"].(string)}
}